lnwallet/parameters.go

```diff
@@ -1,13 +1,50 @@
 package lnwallet
 
 import (
+	"github.com/btcsuite/btcd/mempool"
+	"github.com/btcsuite/btcd/wire"
 	"github.com/btcsuite/btcutil"
-	"github.com/btcsuite/btcwallet/wallet/txrules"
 	"github.com/lightningnetwork/lnd/input"
 )
 
-// DefaultDustLimit is used to calculate the dust HTLC amount which will be
-// send to other node during funding process.
-func DefaultDustLimit() btcutil.Amount {
-	return txrules.GetDustThreshold(input.P2WSHSize, txrules.DefaultRelayFeePerKb)
+// DustLimitForScript retrieves the dust limit for a given pkscript size. Given
+// the size, it automatically determines whether the script is a witness script
+// or not. It calls btcd's GetDustThreshold method under the hood. It must be
+// called with a proper size parameter or else a panic occurs.
+func DustLimitForScript(scriptSize int) btcutil.Amount {
+	var (
+		dustlimit btcutil.Amount
+		pkscript  []byte
+	)
+
+	// With the size of the script, determine which type of pkscript to
+	// create. This will be used in the call to GetDustThreshold. We pass
+	// in an empty byte slice since the contents of the script itself don't
+	// matter.
+	switch scriptSize {
+	case input.P2WPKHSize:
+		pkscript, _ = input.WitnessPubKeyHash([]byte{})
+
+	case input.P2WSHSize:
+		pkscript, _ = input.WitnessScriptHash([]byte{})
+
+	case input.P2SHSize:
+		pkscript, _ = input.GenerateP2SH([]byte{})
+
+	case input.P2PKHSize:
+		pkscript, _ = input.GenerateP2PKH([]byte{})
+
+	case input.UnknownWitnessSize:
+		pkscript, _ = input.GenerateUnknownWitness()
+
+	default:
+		panic("invalid script size")
```

Comment:
Why not just return an error here?

Reply:
o/w the call-sites have to deal with an error, which can't be done with `DefaultBtcChannelConstraints` for example